minmea.c

```diff
@@ -302,10 +302,10 @@ bool minmea_scan(const char *sentence, const char *format, ...)
                         uint32_t value = 0;
                         uint32_t scale = 1000000LU;
                         while (isdigit((unsigned char) *field) && scale > 1) {
-                            value = (value * 10) + (*field++ - '0');
+                            value = (value * 10) + (unsigned char)(*field++ - '0');
                             scale /= 10;
                         }
-                        u = value * scale;
+                        u = (int)(value * scale);
```

Comment:
```suggestion
                        u = (uint32_t)(value * scale);
```

Also, force use `long` for `h`, `i`, `s` variables, as C99 done: https://en.cppreference.com/w/c/string/byte/strtol

Some embedded platforms like MSP430 has 16-bit ints